app/controllers/concerns/parameter_attributes.rb

```diff
@@ -0,0 +1,73 @@
+module ParameterAttributes
+  extend ActiveSupport::Concern
+
+  def process_parameter_attributes
+    model_params = params[parameter_method_mapping]
+    return unless model_params
+    parameter_params = model_params["#{parameter_class_mapping}_parameters_attributes"]
+    return unless parameter_params
+
+    param_names = parameter_params.map { |hash| hash[:name] }
+
+    process_update obj_names, param_names, parameter_params, param_klass
+    process_destroy obj_names, param_names, parameter_params, param_klass
```

Comment:
@xprazak2 I'm curious your thoughts on this in regards to the http://projects.theforeman.org/issues/20725 regression. Is there a good way to fix this so this line of code doesn't delete all parameters?

Reply:
Maybe it could be removed altogether, so update would not remove any existing parameters and could be uses only for update. You would then need to use `hammer host delete-parameter` to remove the parameters.